qemu/tests/hotplug_mem.py

```diff
@@ -31,40 +31,37 @@ def run_background_test(self):
         args = (self.test, self.params, self.env, self.params["sub_type"])
         bg_test = BackgroundTest(run_virt_sub_test, args)
         bg_test.start()
-        wait_for(bg_test.is_alive, timeout=60)
+        wait_for(bg_test.is_alive, first=10, step=3, timeout=100)
         return bg_test
 
-    def restore_memory(self, origin_vm, post_vm):
-        mem_devs_post = post_vm.params.objects("mem_devs")
-        mem_devs_origin = origin_vm.params.objects("mem_devs")
-        if len(mem_devs_post) != len(mem_devs_origin):
-            if len(mem_devs_origin) > len(mem_devs_post):
-                mem_devs = set(mem_devs_origin) - set(mem_devs_post)
-                vm, operation = post_vm, "hotplug"
-            else:
-                mem_devs = set(mem_devs_post) - set(mem_devs_origin)
-                vm, operation = origin_vm, "unplug"
-            func = getattr(self, "%s_memory" % operation)
-            for mem_dev in mem_devs:
-                func(vm, mem_dev)
-                self.check_vm_memory(vm)
+    def restore_memory(self, pre_vm, post_vm):
+        mem_devs_post = set(post_vm.params.objects("mem_devs"))
+        mem_devs_origin = set(pre_vm.params.objects("mem_devs"))
+        if mem_devs_post == mem_devs_origin:
+            return
+        if len(mem_devs_origin) > len(mem_devs_post):
+            mem_devs = mem_devs_origin - mem_devs_post
+            vm, operation = post_vm, "hotplug"
+        elif len(mem_devs_origin) < len(mem_devs_post):
+            mem_devs = mem_devs_post - mem_devs_origin
+            vm, operation = pre_vm, "unplug"
+        func = getattr(self, "%s_memory" % operation)
+        map(lambda x: func(vm, x), mem_devs)
+        vm.verify_alive()
+        vm.reboot()
```

Comment:
Hi, @xutian , Is the reboot here necessary ?  As there is a "reboot action" in "finally" part in start_test function